src/main/dialogs/promotionWindowDialog.js

```diff
@@ -0,0 +1,222 @@
+ * Component that represents the Promotion Window dialog
+ */
+fluid.defaults("gpii.app.promotionWindowDialog", {
+    gradeNames: ["gpii.app.dialog"],
+
+    siteConfig: {
+        promoContentUrl: null,
+        showPromotionWindow: null,
+        showCloseButton: null,
+        width: 200,
+        height: 200,
+        centered: false,
+        resizable: false,
+        movable: false,
+        skipTaskbar: false,
+        frame: false,
+        transparent: false,
+        alwaysOnTop: false,
+        closeDelay: 0,
+        showDelay: 0
+    },
+
+    closeDelay: "{that}.options.siteConfig.closeDelay",
+    showDelay: "{that}.options.siteConfig.showDelay",
+    // some delay, because of slow rearranging of windows taskbar
+    repositionDelay: 1900,
+
+    config: {
+        attrs: {
+            width: "{that}.options.siteConfig.width",
+            height: "{that}.options.siteConfig.height",
+            resizable: "{that}.options.siteConfig.resizable",
+            movable: "{that}.options.siteConfig.resizable.movable",
+            skipTaskbar: "{that}.options.siteConfig.resizable.skipTaskbar",
+            frame: "{that}.options.siteConfig.resizable.frame",
+            transparent: "{that}.options.siteConfig.transparent",
+            alwaysOnTop: "{that}.options.siteConfig.alwaysOnTop"
+        },
+        params: {
+            promoContentUrl: "{that}.options.siteConfig.promoContentUrl",
+            showPromotionWindow: "{that}.options.siteConfig.showPromotionWindow",
+            showCloseButton: "{that}.options.siteConfig.showCloseButton"
+        },
+        fileSuffixPath: "promotionWindow/index.html"
+    },
+
+    events: {
+        delayedClose: null,
+        onRepositioningRequired: null
+    },
+    listeners: {
+        "delayedClose": {
+            funcName: "{closeTimer}.start",
+            args: ["{that}.options.closeDelay"]
+        },
+        onRepositioningRequired: {
+            funcName: "{repositionWindowTimeout}.start",
+            args: ["{that}.options.repositionDelay"]
+        }
+    },
+
+    components: {
+        channelListener: {
+            type: "gpii.app.channelListener",
+            options: {
+                events: {
+                    onPromotionWindowShow: null,
+                    onCloseClicked: null
+                },
+                listeners: {
+                    onPromotionWindowShow: {
+                        funcName: "{promotionWindowDialog}.showPromotionWindow"
+                    },
+                    onCloseClicked: {
+                        funcName: "{promotionWindowDialog}.close"
+                    }
+                }
+            }
+        },
+        closeTimer: {
+            type: "gpii.app.timer",
+            options: {
+                listeners: {
+                    onTimerFinished: {
+                        funcName: "{promotionWindowDialog}.close"
+                    }
+                }
+            }
+        },
+        showTimer: {
```

Comment:
I think we can probably consolidate all of this "show" and "reposition" machinery into one

Reply:
Bump

Reply:
Show and reposition were fixed a long time ago [here](https://github.com/GPII/gpii-app/pull/177/commits/1a8d1e2d8a79cf259fdb3790f31639dfbd015a08) . Let me know if there is anything else that I need to change.